From 0c05ad4a7dd9831464bb82ef474d172928177bb3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 29 Oct 2025 05:37:32 +0100
Subject: [PATCH] Don't hide window on close under Wayland

StatusIcon is not supported on Wayland, causing that a hidden window cannot
be restored. Just quit on close if not running under X11.
---
 src/lxmusic.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/lxmusic.c b/src/lxmusic.c
index e737fcd..e4c20f9 100644
--- a/src/lxmusic.c
+++ b/src/lxmusic.c
@@ -237,7 +237,11 @@ void on_quit(GtkAction* act, gpointer user_data)
 
 gboolean on_main_win_delete_event(GtkWidget* win, GdkEvent* evt, gpointer user_data)
 {
+#if GTK_CHECK_VERSION(3, 0, 0)
+    if( close_to_tray && GDK_IS_X11_DISPLAY(gdk_display_get_default()) )
+#else
     if( close_to_tray )
+#endif
     {
         gtk_widget_hide( win );
         return TRUE;
